Source/Orts.Simulation/Simulation/Signalling/CsSignalScript.cs

```diff
@@ -359,21 +376,87 @@ internal void AttachToHead(SignalHead signalHead)
         /// Called once at initialization time
         /// </summary>
         public abstract void Initialize();
+
         /// <summary>
         /// Called regularly during the simulation
         /// </summary>
         public abstract void Update();
+
         /// <summary>
         /// Called when a signal sends a message to this signal
         /// </summary>
         /// <param name="signalId">Signal ID of the calling signal</param>
         /// <param name="message">Message sent to signal</param>
-        /// <returns></returns>
         public virtual void HandleSignalMessage(int signalId, string message) {}
+
         /// <summary>
         /// Called when the simulator
         /// </summary>
         /// <param name="evt"></param>
         public virtual void HandleEvent(SignalEvent evt, string message = "") { }
+
+        /// <summary>
+        /// Indicates if the signal script has taken control over the speed limit (overrides the speed limit set in the SignalType).
+        /// </summary>
+        public bool SpeedLimitSetByScript { get => SignalHead.SpeedInfoSetBySignalScript; set => SignalHead.SpeedInfoSetBySignalScript = value; }
+
+        /// <summary>
+        /// Returns the parameters of the speed limit set by the script.
+        /// </summary>
+        /// <returns>A tuple containing the parameters of the speed limit, or null if the speed limit is not set</returns>
+        public (float PassengerSpeedLimitMpS, float FreightSpeedLimitMpS, bool Asap, bool Reset, bool NoSpeedReduction, bool IsWarning)? SpeedLimit()
```

Comment:
Is there a reason for using a tuple instead of a struct? I think a struct might be more future-proof in case new flags are added.

Reply:
They are equivalent in terms of future proofing. The only difference is the length of the function definition.

Tuples in C# are structs that don't have a predefined structure.
Here, I used it because it doesn't make really sense creating a new structure just to pass information to another function.

If I change something, I would not use structs but record structs (which are much shorter to declare than structs). But this would require C# 10 which is not yet enabled in the project files.

Reply:
Thanks for your clarification. I had the wrong idea that extending the tuple would break existing scripts.